packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -279,76 +245,21 @@ const SynapseGrid = forwardRef<
   }
 
   const handleChange = (newValue: DataGridRow[], operations: Operation[]) => {
-    if (!modelRef.current) {
+    if (!model) {
       console.error('Model is not initialized')
       return
     }
 
-    // Apply all model updates in one place
-    applyOperationsToModel(operations, newValue, rowValues, modelRef.current)
+    // Check that something changed before updating the model
+    operations = removeNoOpOperations(newValue, rowValues, operations)
 
-    // Handle row tracking for UI state
-    for (const operation of operations) {
-      if (operation.type === 'CREATE') {
-        newValue
-          .slice(operation.fromRowIndex, operation.toRowIndex)
-          .forEach(({ _rowId }: any) => createdRowIds.add(_rowId))
-      }
+    if (operations.length > 0) {
+      // Apply the changes to the model
+      applyOperationsToModel(operations, newValue, rowValues, model)
 
-      if (operation.type === 'UPDATE') {
-        const oldVal = rowValues.slice(
-          operation.fromRowIndex,
-          operation.toRowIndex,
-        )
-        const newVal = newValue.slice(
-          operation.fromRowIndex,
-          operation.toRowIndex,
-        )
-
-        const comparisonResults = oldVal.map((oldItem, idx) =>
-          rowsAreIdentical(oldItem, newVal[idx]),
-        )
-
-        if (comparisonResults.every(Boolean)) return
```

Comment:
Fix a potential bug here where this was an early return if we should ignore the update. If this no-op update was a batch of operations, we would skip all the other operations that we should apply!